prefetch.go

```diff
@@ -951,6 +956,14 @@ func (pgs *PrefetchGlobalState) isDataInCache(
 		// the cached area.
 		return DATA_OUTSIDE_CACHE
 	}
+	// if the end offset is greater than the last byte iov, then we are outside the cache
```

Comment:
@kpjensen why was this bug not triggered more often?  
Is it because prefetching is usually aggressive enough to cover requested byte ranges?

Reply:
I suspect the kernel readahead requests were deemed sequential enough that this rarely occurred.